src/WebJobs.Script/Workers/WorkerConcurrencyManager.cs

```diff
@@ -318,6 +317,26 @@ internal bool IsEnoughMemoryToScale(long hostProcessSize, IEnumerable<long> work
             return true;
         }
 
+        internal bool CanScale(IEnumerable<IRpcWorkerChannel> workerChannels)
+        {
+            // Cancel if there is any "non-ready" channel.
+            var nonreadyWorkerChannels = workerChannels.Where(x => x.IsChannelReadyForInvocations() == false);
+            if (nonreadyWorkerChannels.Any())
+            {
+                _logger.LogDebug($"Starting new language worker canceled: TotalChannels={workerChannels.Count()}, NonReadyChannels={nonreadyWorkerChannels.Count()}");
+                return false;
+            }
+
+            // Cancel if MaxWorkerCount is reached.
+            if (workerChannels.Count() >= _workerConcurrencyOptions.Value.MaxWorkerCount)
```

Comment:
Instead of executing twice, store the result of `Count()` call to a variable and use it in 331 and 333.